src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/queryAllPackages.hpp

```diff
@@ -0,0 +1,254 @@
+/*
+ * Wazuh Vulnerability scanner - Scan Orchestrator
+ * Copyright (C) 2015, Wazuh Inc.
+ * Jan 22, 2024.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#ifndef __QUERY_ALL_PKGS_FOR_AGT_HPP
+#define __QUERY_ALL_PKGS_FOR_AGT_HPP
+
+#include "chainOfResponsability.hpp"
+#include "loggerHelper.h"
+#include "scanContext.hpp"
+#include "socketDBWrapper.hpp"
+#include "vulnerabilityScanner.hpp"
+#include "wazuhDBQueryBuilder.hpp"
+
+/**
+ * @brief Orchestrates queries to the global Wazuh-DB system and initiates package scanning.
+ *
+ * The `TQueryAllPackages` class is responsible for coordinating the execution of queries within the
+ * broader Wazuh ecosystem.
+ *
+ * @tparam TScanContext scan context type.
+ */
+template<typename TScanContext = ScanContext>
+class TQueryAllPackages final : public AbstractHandler<std::shared_ptr<TScanContext>>
+{
+private:
+    std::shared_ptr<AbstractHandler<std::shared_ptr<TScanContext>>> m_suborchestration;
+    std::optional<SocketDBWrapper> m_wdbSocketWrapper {std::nullopt};
+
+public:
+    /**
+     * @brief Construct a new global fetch object
+     *
+     * @param packageInsertOrchestration package orchestration instance.
+     */
+    explicit TQueryAllPackages(
+        std::shared_ptr<AbstractHandler<std::shared_ptr<TScanContext>>> packageInsertOrchestration)
+        : m_suborchestration(std::move(packageInsertOrchestration))
+    {
+    }
+
+    /**
+     * @brief Handles request and passes control to the next step of the chain.
+     *
+     * This method processes a request based on the operation type specified in the input data.
+     * If the operation type is SingleDeleteAndInsert, it queries a single entry from the Wazuh-DB and processes the
+     * response. If the operation type is InsertAll, it executes queries for all packages from the Wazuh-DB and
+     * processes the responses. If the operation type is not recognized, it throws a std::runtime_error.
+     *
+     * @param data A shared pointer to the input data.
+     * @return A shared pointer to the result of the request processing.
+     *
+     * @note This function interacts with a Wazuh-DB socket, processes JSON responses, and creates and handles data
+     * using flatbuffers. It may also throw exceptions if certain conditions are not met.
+     */
+    std::shared_ptr<TScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
+    {
+        // Instance the socketWrapper for Wazuh-DB
+        if (!m_wdbSocketWrapper)
```

Comment:
Instance WDB for each handleRequest call, members on the chain class will be dangerous, because this class is used by multiple threads.

Reply:
Moved inside of method `handleRequest`. Thx!